superset-frontend/src/views/CRUD/utils.tsx

```diff
@@ -380,3 +393,57 @@ export const hasTerminalValidation = (errors: Record<string, any>[]) =>
             isNeedsPassword(payload) || isAlreadyExists(payload),
         ),
   );
+
+function buildDashboardTabsTree(
+  nodeId: string,
+  data: { [key: string]: any },
+  tree: any,
+  traversedIds: any,
```

Comment:
can we type these better? It looks like `tree` is an array of some sort, and traversedIds is a `Record<string, any>`, but maybe we can do even better than my cursory glance at typing